wordpress-mobile/WordPress-Android
# Change pin/hide/remove to site scope

Fixes #14018

Related FluxC changes - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1868

This PR adds the functionality to pin/unpin/hide/remove cards based on the selected site. The main change is that the logic and the models were moved to FluxC.

For all tests:
- Make sure you turn on MySiteImprovements flag and restart the app
- Make sure you have 2 sites with QuickStart started

To test (pin):
- Go to a site A
- Pin the "Grow" card
- Notice it jumps to the first place
- Switch site to site B
- Notice the "Customize" card is in the first place
- Switch back to site A
- Notice the "Grow" card is still in the first place
- Restart the app
- Notice the "Grow" card is still in the first place
- Unpin the "Grow" card
- Notice it jumps back to the second place


To test (remove):
- Go to a site A
- Remove a card A
- Notice it disappears 
- Switch site to site B
- Notice the card A is visible
- Switch back to site A
- Notice the card A is still gone
- Restart the app
- Notice the card A is still gone


To test (hide):
- Go to a site A
- Hide a card A
- Notice it disappears 
- Switch site to site B
- Notice the card A is visible
- Switch back to site A
- Notice the card A is still gone
- Restart the app
- Notice the card A is visible again

PR submission checklist:
